compiler/src/dotty/tools/dotc/config/CliCommand.scala

```diff
@@ -138,11 +138,37 @@ trait CliCommand:
   protected def yusageMessage(using settings: ConcreteSettings)(using SettingsState) =
     createUsageMsg("Possible private", shouldExplain = true, isPrivate)
 
+  /** Used for the formatted output of -Xshow-phases */
   protected def phasesMessage: String =
-    (new Compiler()).phases.map {
-      case List(single) => single.phaseName
-      case more => more.map(_.phaseName).mkString("{", ", ", "}")
-    }.mkString("\n")
+    val phases = new Compiler().phases
+    val nameLimit = 25 
+    val maxCol = 100
```

Comment:
If you have access to a Context here you could use `ctx.settings.pageWidth.value` which should default to the column width of the terminal.

Reply:
Yeah I think ideally sbt should pass -pagewidth to the compiler (or ensure $COLUMN is set correctly). Can you just let it wrap over instead of truncating?

Reply:
I think actually just making max way longer is better than wrapping here. Then the user can let the terminal handle the wrap or whatever happens there instead of trying to code in wrapping functionality.